### ISSUE TITLE
[Feature] Upgrade Teardown content creation ecosystem to next level

### ISSUE THREAD

**Issue by trsh** [Teardown-Issue-Tracker-Maintainers/Teardown-Issue-Tracker#320]
### Is your feature request related to a problem (if not, explain why you think this feature should be added)? Please describe.

I have been now playing with the modding tools for some while. And I can just say almost everything about it feels fantastic. I would say that its even something more the modding. It feels like real game creation tool like Roblox. There are two missing pieces, animation and networking, that as I saw, are already in development.

However the current state of modding doesn't really suitable for some individuals and teams like mine. Creating/coding is fun for us, but we also like to make living from it, to keep going. This currently is not possible because:

- There is no protection of intellectual property, all scripts are available for everyone in game directory and a month long work can just be copy pasted.
- The 'Mod' content browser needs a upgrade to promote most relevant games.
-  No monetization options.

### Describe the solution you'd like

I would like to propose to take Teardown to something like motioned Roblox or similar:

- Change the name from mods, to games and items and have option to categorize those
- The game index page could feature advanced game and item browser, with search box, list of top games, top rated, recommended, etc. Inhouse games pined on top if wish so.
- Creators would have option to make sources private (on by default).
- There could be a in game currency. 
- Creators could ask for small one time entry fee for their level and have in game purchases via special provided API.

Pros:

- Teardown team could take a small fee from all in-game-purchases etc. and grow bigger financially, even make the game free at 1 point.
- Product will become much more popular. 
- Small teams like mine will jump in to bring HQ games.
- Individuals can also benefit and spend more time on TD games.
- There will be so much content to play.

Cons:

- This will require investment and expertise beyond game making.
- Server requirements will grow with more content and more players.

Neutral:

- Small studios tend to put Indy modders/creators a bit into shadows. But I say its good thing. Playing along with bigger players promotes will to grow and succeed. I think the modders will still have same or more fans with so much more players in the ecosystem. Also what Roblox proves again and again, that indie devs can succeed and even outrun studios if the game idea is on point.

### Describe alternatives you've considered

It was suggested to put Patreon link in description, which is neat idea, but on totally different level from what I am proposing here. 

### Additional context

Please don't throw rotten tomatoes at me if the FR seams inappropriate :D

**Comment by YuLun-bili** [comment]
You do came up with some good suggestions, but monetization is the worst one ever suggested.

You pointed out 3 issues related:
* protection of intellectual property
* upgraded mod browser
* monetization

First 2 issues are very good, they even deserve being two individual issues. Third one is just ridiculous.

> Because we code for fun but also make living from it, to keep going.

If you choose to mod for fun, not as career, it's totally impossible to make living from it. "Mod for fun" determines one must already be able to live without any difficulties in any aspects (e.g.: financially) and have enough free time and energy. Any earnings form "mod for fun" could only improve their life finitely. Therefore this doesn't make sense.

> Teardown team can take a small fee from all purchases and grow bigger, even make the game free at 1 point.

According to [this press release in August 2022](https://embracer.com/releases/embracer-group-acquires-tuxedo-labs/), this "small income" is negligible. And I'm sure that making the game free would never happen in predictable future.

Additionally, monetization mods is already possible via third party platforms. We have Patron, PayPal Donate, etc. It's much more easier and practical to use these platforms rather than suggesting game to build one for mod creators.

*Throws fresh tomatoes*

**Comment by trsh** [comment]
> If you choose to mod for fun, not as career, it's totally impossible to make living from it. "Mod for fun" determines one must already be able to live without any difficulties in any aspects (e.g.: financially) and have enough free time and energy. Any earnings form "mod for fun" could only improve their life finitely. Therefore this doesn't make sense.

I don't think we should be here debating how correct I formed my idea. Me and my group like making games/modes/levels/props. Its Fun for us (i.e we enjoy doing it), but we also get compensated and can keep doing these things. That was my only point.

> According to [this press release in August 2022](https://embracer.com/releases/embracer-group-acquires-tuxedo-labs/), this "small income" is negligible. And I'm sure that making the game free would never happen in predictable future.

What I said has nothing todo with August 2022 and predictable future. I am talking about the I-WISH future where all of what I am saying would be implemented, and then TD would take % from in-game-purchases etc. In general this schema works for lot's of platforms. It's not something new.

> Additionally, monetization mods is already possible via third party platforms. We have Patron, PayPal Donate, etc. It's much more easier and practical to use these platforms rather than suggesting game to build one for mod creators.

This is not monetization provided from TD and I would not even call it this way. I have been explained that best I can do (by not violating the rules), is to put a donation link in my mod description. The mod Lua is restricted, and doesn't allow me to call 3d party services for good reason. And sorry, donation link is just not enough to survive. 

_These tomatoes go back_

_P.S. I fixed some parts, as they seem to create confusion_
